client/session.go

```diff
@@ -1252,6 +1263,7 @@ func (r *blocksResult) addBlockFromPeer(id string, block *rpc.Block) error {
 		defer iter.Close()
 
 		encoder := r.encoderPool.Get()
+		encoder.Reset(start, r.blockAllocSize)
 		defer encoder.Close()
```

Comment:
Yeah probably not lol, will update and also assert not closed by pulling the test encoder out and checking it's state.